app/assets/javascripts/two-pane.js

```diff
@@ -144,6 +144,11 @@ function right_pane_content(response){
   var contentId;
   if (handle_redirect(response)) return; //session expired redirect to login
 
+  // move flash messages from response to main window
+  $('.flash', response).each(function(index, item) {
```

Comment:
Trying to translate this to the develop branch, i ended up with someting like this:
```
function right_pane_content(response){
  var contentId;
  if (handle_redirect(response)) return; //session expired redirect to login

  // move notifications to main document
  var notification_tag = $('#notifications', response)[0];
  if (notification_tag) {
    var dest_array = $($('#notifications')[0]).data('flash');
    var flash_array = $(notification_tag).data('flash');
    var length = flash_array.length;
    var i;
    $(notification_tag).detach();
    if (dest_array == "") {
      dest_array = Array();
    }
    for (i = 0; i < length; i++){
      dest_array.push(flash_array[i]);
    }
    $('#notifications').data('flash', dest_array);
    tfm.toastNotifications.clear();
  }

  if (!$("#content", response).length){
<...>
```
If we could be sure, that the earlier notifications were already toasted, it might suffice to just replace the whole `data-flash` attribute.